docs/native-components-android.md

```diff
@@ -273,8 +488,121 @@ public class MyFragment extends Fragment {
 }
 ```
 
+</TabItem>
+</Tabs>
+
 ### 3. Create the `ViewManager` subclass
 
+<Tabs groupId="android-language" defaultValue={constants.defaultAndroidLanguage} values={constants.androidLanguages}>
+<TabItem value="kotlin">
+
+```kotlin title="MyViewManager.kt"
+// replace with your package
+package com.mypackage
+
+import android.view.Choreographer
+import android.view.View
+import android.view.ViewGroup
+import android.widget.FrameLayout
+import androidx.fragment.app.FragmentActivity
+import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.bridge.ReadableArray
+import com.facebook.react.uimanager.ThemedReactContext
+import com.facebook.react.uimanager.ViewGroupManager
+import com.facebook.react.uimanager.annotations.ReactPropGroup
+import kotlin.properties.Delegates
+
+class MyViewManager(
+    private val reactContext: ReactApplicationContext
+) : ViewGroupManager<FrameLayout>() {
+  private var propWidth by Delegates.notNull<Int>()
+  private var propHeight by Delegates.notNull<Int>()
+
+  override fun getName() = REACT_CLASS
+
+  /**
+   * Return a FrameLayout which will later hold the Fragment
+   */
+  override fun createViewInstance(reactContext: ThemedReactContext) =
+      FrameLayout(reactContext)
+
+  /**
+   * Map the "create" command to an integer
+   */
+  override fun getCommandsMap() = mapOf("create" to COMMAND_CREATE)
+
+  /**
+   * Handle "create" command (called from JS) and call createFragment method
+   */
+  override fun receiveCommand(
+      root: FrameLayout,
+      commandId: String,
+      args: ReadableArray?
+  ) {
+    super.receiveCommand(root, commandId, args)
+    val reactNativeViewId = requireNotNull(args).getInt(0)
+
+    when (commandId.toInt()) {
+      COMMAND_CREATE -> createFragment(root, reactNativeViewId)
+    }
+  }
+
+  @ReactPropGroup(names = ["width", "height"], customType = "Style")
+  fun setStyle(view: FrameLayout, index: Int, value: Int) {
+    if (index == 0) propWidth = value
+    if (index == 1) propHeight = value
+  }
+
+  /**
+   * Replace your React Native view with a custom fragment
+   */
+  fun createFragment(root: FrameLayout, reactNativeViewId: Int) {
+    val parentView = root.findViewById<ViewGroup>(reactNativeViewId)
+    setupLayout(parentView)
+
+    val myFragment = MyFragment()
+    val activity = reactContext.currentActivity as FragmentActivity
+    activity.supportFragmentManager
+        .beginTransaction()
+        .replace(reactNativeViewId, myFragment, reactNativeViewId.toString())
+        .commit()
+  }
+
+  fun setupLayout(view: View) {
+    Choreographer.getInstance().postFrameCallback(object: Choreographer.FrameCallback {
```

Comment:
I believe here we could use SAM conversion (i.e. you don't need `object` and the `override` method declaration.

Reply:
That was my thought as well but it causes issues with scoping of `this`. We call `Choreographer.getInstance().postFrameCallback(this)` inside the lambda. `this` will reference our `MyViewManager` instead of the `Choreographer.FrameCallback`.

So unfortunately we need to use the `object` syntax in this case so that `this` is scoped correctly.

![image](https://user-images.githubusercontent.com/7195844/163731433-ceb0e7a1-8e37-44ef-a276-fda459ea8763.png)

Reply:
Might be possible through some other syntax though so be happy to change it into a lambda if you can find one. I couldn't though.

Reply:
You're right. As this is a java interface, we'll have to use this Compiler Plugin here:
https://kotlinlang.org/docs/sam-with-receiver-plugin.html
(which we can't assume our user are enabling)